### ISSUE TITLE
Segmentation faults occur very often

### ISSUE THREAD

**Issue by Artuto** [clitic/vsd#6]
Hi, I'm encountering many segmentation fault crashes very very often (it happens 9/10 runs).

Using this command:

`vsd save "url" -o file.mp4 --key [redacted key]`

This is the usual output:

```
Run 1:
✔ Select one variant stream: · 720p (3.55 MB/s)
  Encrypted CENC streams found in playlist
Downloading video stream to (video) default (1).m4s
        Key [REDACTED]
 Downloaded video stream successfully
Downloading audio stream to (audio) default.m4s
        Key [REDACTED]
⠦ 2.67 / 2.78 GB ━━━━━━━━━━━━━━╸━━━━━━━━━━━━━━ 50.00% • 4591/9182 • 03:08 > 03:08 • 24.35 SEG/s • 14.54 MB/s
Segmentation fault (core dumped)

Run 2:
✔ Select one variant stream: · 720p (3.55 MB/s)
  Encrypted CENC streams found in playlist
Downloading video stream to (video) default (1).m4s
        Key [REDACTED]
 Downloaded video stream successfully
Downloading audio stream to (audio) default.m4s
        Key [REDACTED]
⠦ 2.67 / 2.78 GB ━━━━━━━━━━━━━━╸━━━━━━━━━━━━━━ 50.00% • 4591/9182 • 03:08 > 03:08 • 24.35 SEG/s • 14.54 MB/s
Segmentation fault (core dumped)

Run 3:
✔ Select one variant stream: · 720p (3.55 MB/s)
  Encrypted CENC streams found in playlist
Downloading video stream to (video) default (2).m4s
        Key [REDACTED]
⠏ 2.43 / 2.76 GB ━━━━━━━━━━━━━╸━━━━━━━━━━━━━━━ 45.76% • 4202/9182 • 02:19 > 02:45 • 30.03 SEG/s • 17.90 MB/s
Segmentation fault (core dumped)

Run 4:
✔ Select one variant stream: · 720p (3.55 MB/s)
  Encrypted CENC streams found in playlist
Downloading video stream to (video) default (3).m4s
        Key [REDACTED]
 Downloaded video stream successfully
Downloading audio stream to (audio) default (1).m4s
        Key [REDACTED]
⠦ 1.33 / 1.40 GB ━━━━━━━━━━━━━━━━━╸━━━━━━━━━━━━ 59.61% • 5473/9182 • 02:03 > 01:23 • 44.23 SEG/s • 2.57 MB/s
Segmentation fault (core dumped)
```

**Comment by clitic** [comment]
@Artuto What is your operating system?
Does the downloaded video is decrypted and playable?

You can try

1. Limiting the threads `-t 1`.
2. Using key in `KID:KEY` format.

**Comment by Artuto** [comment]
Using Ubuntu 20.04 on WSL (Windows 10 64bits)

Will try limiting threads.

**Comment by clitic** [comment]
Maybe there are some memory issues I would recommend 
you to use windows [binary](https://github.com/clitic/vsd/releases/download/v0.2.0/vsd-v0.2.0-x86_64-pc-windows-msvc.zip).

> Does the downloaded video is decrypted and playable?

**Comment by Artuto** [comment]
I tried setting the treads to 2 and that seems to be safe so it doesnt crash anymore. Tried the windows binary and still had many issues.

> Does the downloaded video is decrypted and playable?

yes it is

**Comment by clitic** [comment]
@Artuto Issue is fixed by 2aa9468.
